pkg/compliance/checks/docker.go

```diff
@@ -0,0 +1,31 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-2020 Datadog, Inc.
+
+// +build docker
+
+package checks
+
+import (
+	"context"
+	"time"
+
+	"github.com/DataDog/datadog-agent/pkg/config"
+	"github.com/DataDog/datadog-agent/pkg/util/docker"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+func newDockerClient() DockerClient {
+	queryTimeout := config.Datadog.GetDuration("docker_query_timeout") * time.Second
+
+	// Major failure risk is here, do that first
+	ctx, cancel := context.WithTimeout(context.Background(), queryTimeout)
+	defer cancel()
+	client, err := docker.ConnectToDocker(ctx)
```

Comment:
why don't we use `GetDockerUtil` just like all the other packages that need the docker client?
we're missing out the retry logic that `GetDockerUtil` provides...

Reply:
Personally I found DockerUtil to be quite confusing - it obfuscates the use of the client and the initialization/singleton use is far from idea. I suspect the goal there was to shield the client from exposure elsewhere in the code. Additionally using `GetDockerUtil` would result  in complicating mocks it tests here. I will look into the retry logic to see if we can externalize it.

Reply:
we can consider building a "lighter" docker util in the package docker that just provides the client with a retrying logic